lib/Dialect/FIRRTL/LowerToRTL.cpp

```diff
@@ -97,6 +97,29 @@ static LogicalResult lower(firrtl::ConstantOp op, ArrayRef<Value> operands,
   return success();
 }
 
+static LogicalResult lower(firrtl::WireOp op, ArrayRef<Value> operands,
+                           ConversionPatternRewriter &rewriter) {
+  auto resType = op.result().getType();
+  if (auto resFirType = resType.dyn_cast<FIRRTLType>()) {
```

Comment:
The type of `firrtl::WireOp` is always a FIRRTLType, so you can eliminate the if and simplify this to:
```
auto resType = op.result().getType().cast<FIRRTLType>();
```

Incidentally, and separate from this patch, I would love to figure out how to make `getType()` return a FIRRTLType directly, as that would make working with the FIRRTL dialect a lot nicer.  I think there is probably a way to do that in the firrtl .td files, I filed https://github.com/circt/circt/issues/17 to track this.